pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -203,7 +217,7 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 	} else {
 		// For non-attachable volumes, generate a unique name based on the pod
 		// namespace and name and the name of the volume within the pod.
-		volumeName = volumehelper.GetUniqueVolumeNameForNonAttachableVolume(podName, volumePlugin, outerVolumeSpecName)
+		volumeName = volumehelper.GetUniqueVolumeNameForNonAttachableVolume(podName, volumePlugin, volumeSpec)
```

Comment:
After we change the logic to only use pod uid to check whether the desired or actual states contains the information about pod volume, using outerVolumeSpecName should also ok. But using outerVolumeSpecName is kind of confusing. It seems to me more clear to pass volumeSpec instead of a specific type of name in this function.